packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -149,15 +156,29 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
     );
 
     // Retrieve Unique ID
-    final savedFixedEndpointId =
-        await _endpointInfoStore.read(key: endpointIdStorageKey);
-    final fixedEndpointId = savedFixedEndpointId ?? const Uuid().v1();
-    if (savedFixedEndpointId == null) {
+    final endpointInformationVersion =
+        await _endpointInfoStore.read(key: _endpointInformationVersionKey);
+    String? retrievedEndpointId;
+
+    if (endpointInformationVersion == null) {
+      _endpointInfoStore.write(
+        key: _endpointInformationVersionKey,
+        value: _endpointInformationCurrentVersionValue,
+      );
+
+      await _legacyNativeDataProvider?.initialize(appId);
+      retrievedEndpointId = await _legacyNativeDataProvider?.getEndpointId();
+
+      retrievedEndpointId ??= const Uuid().v1();
```

Comment:
question: Why use this version of uuid instead of `uuid()` from aws_common? If there is specific requirement for that could we have a comment here saying why?

Reply:
Thanks there is no reason, wasn't aware of this method in core package.